src/app/modules/myinfo/myinfo.util.ts

```diff
@@ -333,80 +359,143 @@ export const isMyInfoCookie = (
-  cookie.state === MyInfoCookieState.Success
+export const assertOldMyInfoCookieSuccessState = (
+  cookie: MyInfoOldCookiePayload,
+): Result<MyInfoSuccessfulOldCookiePayload, MyInfoCookieStateError> =>
+  cookie.state === MyInfoOldCookieState.Success
     ? ok(cookie)
     : err(new MyInfoCookieStateError())
 
 /**
+ * TODO(#5452): Delete this function
  * Extracts and asserts a successful myInfoCookie from a request's cookies
- * @param cookies Cookies in a request
+   @param cookies Cookies in a request
  * @return ok(cookie) the successful myInfoCookie
  * @return err(MyInfoMissingAccessTokenError) if myInfoCookie is not present on the request
  * @return err(MyInfoCookieStateError) if the extracted myInfoCookie was in an error state
  * @return err(MyInfoCookieAccessError) if the cookie has been accessed before
- */
-export const extractAndAssertMyInfoCookieValidity = (
+  */
+export const extractAndAssertOldMyInfoCookieValidity = (
   cookies: Record<string, unknown>,
 ): Result<
-  MyInfoSuccessfulCookiePayload,
-  | MyInfoCookieStateError
-  | MyInfoMissingAccessTokenError
-  | MyInfoCookieAccessError
+  MyInfoSuccessfulOldCookiePayload,
+  MyInfoCookieStateError | MyInfoMissingAccessTokenError
 > =>
-  extractMyInfoCookie(cookies)
-    .andThen((cookiePayload) => assertMyInfoCookieSuccessState(cookiePayload))
-    .andThen((cookiePayload) => assertMyInfoCookieUnused(cookiePayload))
-
-/**
- * Asserts that the myInfoCookie has not been used before
- * @param myInfoCookie
- * @returns ok(myInfoCookie) if the cookie has not been used before
- * @returns err(MyInfoCookieAccessError) if the cookie has been used before
- */
-export const assertMyInfoCookieUnused = (
-  myInfoCookie: MyInfoSuccessfulCookiePayload,
-): Result<MyInfoSuccessfulCookiePayload, MyInfoCookieAccessError> =>
-  myInfoCookie.usedCount <= 0
-    ? ok(myInfoCookie)
-    : err(new MyInfoCookieAccessError())
+  extractOldMyInfoCookie(cookies).andThen((cookiePayload) =>
+    assertOldMyInfoCookieSuccessState(cookiePayload),
+  )
 
 /**
+ * TODO(#5452): Delete this function
  * Extracts access token from a MyInfo cookie
  * @param cookie Cookie from which access token should be extracted
  */
-export const extractAccessTokenFromCookie = (
-  cookie: MyInfoCookiePayload,
+export const extractAccessTokenFromOldCookie = (
+  cookie: MyInfoOldCookiePayload,
 ): Result<string, MyInfoCookieStateError> => {
-  if (cookie.state !== MyInfoCookieState.Success) {
+  if (cookie.state !== MyInfoOldCookieState.Success) {
     return err(new MyInfoCookieStateError())
   }
   return ok(cookie.accessToken)
 }
 
+/**
+ * Extracts a MyInfo login cookie from a request's cookies
+ * @param cookies Cookies in a request
+ */
+export const extractMyInfoLoginJwt = (
+  cookies: Record<string, unknown>,
+): Result<string, MyInfoMissingLoginCookieError> => {
+  const jwt = cookies[MYINFO_LOGIN_COOKIE_NAME]
+  if (typeof jwt === 'string') {
```

Comment:
should we also check whether the string is empty? 
```suggestion
  if (typeof jwt === 'string' && !!jwt) {
```

Reply:
yup sure, catches the error earlier in the chain

Reply:
676072bc